app/helpers/template_helpers.py

```diff
@@ -144,9 +144,28 @@ def footer_context(self) -> dict[str, Any]:
             context["footerWarning"] = self._footer_warning
 
         if self._survey_config.footer_links:
-            context["rows"] = [{"itemsList": self._survey_config.footer_links}]
-
-        if self._survey_config.footer_legal_links:
+            if self._survey_type:
+                context["rows"] = [{"itemsList": self._survey_config.footer_links}]
+
+            else:
+                items_list: list[Mapping] = []
+
+                for footer_link in self._survey_config.footer_links:
```

Comment:
The survey config class should be responsible for returning the links it needs instead of it being handled externally by template helpers. Instead of filtering out stuff we don't need, we should populate with what we need from the get go, which you can, if the logic was handled by the survey config object. I would convert the attributes to methods which gets passed in a `bool` to whether you have a cookie theme, similar to how `get_service_links` is done.

Reply:
Ok, moved all the links generation similar to `get_service_links`, added extra test for Welsh census for coverage, have a look!